substrate-archive-common/src/util.rs

```diff
@@ -138,3 +108,37 @@ fn format_opt(file: Option<String>) -> String {
         Some(f) => f,
     }
 }
+
+/// Get the path to a local substrate directory where we can save data.
+/// Platform | Value | Example
+/// -- | -- | --
+/// Linux | $XDG_DATA_HOME or $HOME/.local/share/substrate_archive | /home/alice/.local/share/substrate_archive/
+/// macOS | $HOME/Library/Application Support/substrate_archive | /Users/Alice/Library/Application Support/substrate_archive/
+/// Windows | {FOLDERID_LocalAppData}\substrate_archive | C:\Users\Alice\AppData\Local\substrate_archive
+pub fn substrate_dir() -> Result<PathBuf, &'static str> {
+    if let Some(base_dirs) = dirs::BaseDirs::new() {
+        let mut path = base_dirs.data_local_dir().to_path_buf();
+        path.push("substrate_archive");
+        Ok(path)
+    } else {
+        Err("No valid home directory path could be retrieved from the operating system")
+    }
+}
+
+/// Create an arbitrary directory on disk.
+pub fn create_dir(path: &Path) -> Result<(), &'static str> {
+    if let Err(e) = std::fs::create_dir_all(path) {
+        match e.kind() {
+            std::io::ErrorKind::AlreadyExists => (),
+            _ => return Err("A directory in path could not be created"),
```

Comment:
It would be better to capture the error and return it, worst case as `Err(format!("oh noes, badness: {:?}", e))`.

Reply:
#122